File: backend/src/main/java/cl/helpcom/facturacion/proyectos/dominio/Proyecto.java
package cl.helpcom.facturacion.proyectos.dominio;

import cl.helpcom.facturacion.clientes.dominio.Cliente;
import cl.helpcom.facturacion.clientes.dominio.TipoServicio;
import cl.helpcom.facturacion.comun.dominio.EntidadAuditable;
import cl.helpcom.facturacion.comun.dominio.Moneda;
import cl.helpcom.facturacion.empresa.dominio.Empresa;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.time.LocalDate;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "proyecto")
@Getter
@Setter
@NoArgsConstructor
public class Proyecto extends EntidadAuditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "empresa_id", nullable = false)
    private Empresa empresa;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tipo_servicio_id")
    private TipoServicio tipoServicio;

    @Column(name = "codigo", length = 40)
    private String codigo;

    @Column(name = "nombre", nullable = false, length = 200)
    private String nombre;

    @Column(name = "descripcion", length = 500)
    private String descripcion;

    @Column(name = "precio_base_neto", nullable = false, precision = 18, scale = 4)
    private BigDecimal precioBaseNeto;

    @Enumerated(EnumType.STRING)
    @Column(name = "moneda_precio", nullable = false, length = 3)
    private Moneda monedaPrecio;

    @Enumerated(EnumType.STRING)
    @Column(name = "periodicidad", nullable = false, length = 10)
    private Periodicidad periodicidad;

    @Column(name = "dia_facturacion", nullable = false)
    private Short diaFacturacion;

    @Column(name = "fecha_inicio", nullable = false)
    private LocalDate fechaInicio;

    @Column(name = "fecha_termino")
    private LocalDate fechaTermino;

    @Column(name = "activo", nullable = false)
    private boolean activo = true;
}

File: frontend/types/proyecto.ts
import type { Moneda, Periodicidad } from "./dominio";

/** Espeja `ProyectoRespuestaDto` (backend proyectos/dto/ProyectoRespuestaDto.java). */
export interface Proyecto {
  id: number;
  clienteId: number;
  clienteRazonSocial: string;
  tipoServicioId: number | null;
  tipoServicioNombre: string | null;
  codigo: string | null;
  nombre: string;
  descripcion: string | null;
  /** Neto, sin IVA, en `monedaPrecio`. */
  precioBaseNeto: number;
  monedaPrecio: Moneda;
  periodicidad: Periodicidad;
  diaFacturacion: number;
  /** ISO-8601 (AAAA-MM-DD). */
  fechaInicio: string;
  fechaTermino: string | null;
  activo: boolean;
}

/** Espeja `ProyectoSolicitudDto`. */
export interface ProyectoSolicitud {
  clienteId: number;
  tipoServicioId: number | null;
  codigo: string | null;
  nombre: string;
  descripcion: string | null;
  precioBaseNeto: number;
  monedaPrecio: Moneda;
  periodicidad: Periodicidad;
  diaFacturacion: number;
  fechaInicio: string;
  fechaTermino: string | null;
  activo: boolean;
}
